src/routes/CustomAsset/components/CustomAsset.js

```diff
@@ -2,110 +2,90 @@
 
 import React from 'react'
 
-import { JInput, JRaisedButton } from 'components/base'
-import { ButtonWithConfirm, ModalHeader } from 'components'
+import { JInput, JRaisedButton } from '../../../components/base'
+import { ModalHeader } from '../../../components'
+
+type CustomAssetFormType = 'add' | 'edit'
+
+type CustomAssetProps = {
+  formFields: CustomAssetFormFields,
+  invalidFields: CustomAssetFormFields,
+  setField: SetFieldFunction<CustomAssetFormFields>,
+  submit: () => void,
+
+  // optional
+  isAssetLoading: boolean,
+  type: CustomAssetFormType,
+}
 
 const CustomAsset = ({
-  add,
-  edit,
-  remove,
-  setName,
-  setSymbol,
-  setAddress,
-  setDecimals,
+  formFields,
   invalidFields,
-  name,
-  symbol,
-  address,
-  decimals,
+  submit,
+  setField,
+  isAssetLoading,
   type,
-}: Props) => {
-  if (!['add', 'edit'].includes(type)) {
-    return null
-  }
+}: CustomAssetProps) => {
+  const setFieldHandler = (fieldName: $Keys<CustomAssetFormFields>) =>
+    (value: string) => setField(fieldName, value)
 
-  const isAdd: boolean = (type === 'add')
+  const isAddressFieldDisabled = (type === 'edit')
+  const i18nKey = (type === 'add') ? 'addCustomAsset' : 'editCustomAsset'
 
   const fields: Array<{
-    handler: Function,
-    key: string,
-    value: string,
+    key: $Keys<CustomAssetFormFields>,
     isDisabled: boolean,
+    isLoading: boolean,
   }> = [
-    { handler: setAddress, key: 'address', value: address, isDisabled: !isAdd },
-    { handler: setName, key: 'name', value: name, isDisabled: false },
-    { handler: setSymbol, key: 'symbol', value: symbol, isDisabled: false },
-    { handler: setDecimals, key: 'decimals', value: decimals, isDisabled: false },
+    { key: 'address', isDisabled: isAddressFieldDisabled, isLoading: isAssetLoading },
+    { key: 'name', isDisabled: false, isLoading: false },
+    { key: 'symbol', isDisabled: false, isLoading: false },
+    { key: 'decimals', isDisabled: false, isLoading: false },
   ]
 
   return (
     <div className='custom-asset'>
       <div className='header'>
         <ModalHeader
-          title={`${isAdd ? 'Add' : 'Edit'} Custom Asset`}
+          title={`routes.${i18nKey}.title`}
           color='gray'
           withMenu
         />
       </div>
       <div className='form'>
```

Comment:
I would definitely move form to its own component and re-use it in edit and create something like
```
<AssetForm model={}/> in case of new
<AssetForm model={asset} in case of edit
```
You can do it in the next PR with editing